src/graphql/graphql.ts

```diff
@@ -1481,6 +1481,94 @@ export function toIncrementalV2(query: string): string {
   return gql.print(ast);
 }
 
+/**
+ * Extracts the model from a V1 query
+ *
+ * Example, for query:
+ *  vcs {
+ *     pullRequests {
+ *       nodes {
+ *         title
+ *       }
+ *     }
+ *  }
+ *
+ * returns:
+ * {
+ *  namespace: 'vcs',
+ *  name: 'PullRequest',
+ *  fullName: 'vcs_PullRequest'
+ * }
+ */
+export function extractModelNameV1(query: string): Model {
+  const fields: string[] = [];
+
+  gql.visit(gql.parse(query), {
+    Field: {
+      enter(node) {
+        const name = node.alias?.value ?? node.name.value;
+
+        if (name === NODES) {
+          return false;
+        }
+
+        fields.push(name);
+        return undefined;
+      },
+    },
+  });
+
+  ok(fields.length === 2, `expected 2 elements in ${fields}`);
+
+  const namespace = fields[0];
+  const name = _.upperFirst(singular(fields[1]));
+  return {
+    namespace,
+    name,
+    fullName: [namespace, name].join('_'),
+  };
+}
+
+/**
+ * Extracts the model from a V2 query
+ *
+ * Example, for query:
+ *  vcs_PullRequest {
+ *    title
+ *  }
+ *
+ * returns:
+ * {
+ *  namespace: 'vcs',
+ *  name: 'PullRequest',
+ *  fullName: 'vcs_PullRequest'
+ * }
+ */
+export function extractModelNameV2(query: string): Model {
+  let firstField: string | undefined;
+
+  gql.visit(gql.parse(query), {
+    Field: {
+      enter(node) {
+        firstField = node.alias?.value ?? node.name.value;
+        return false;
+      },
+    },
+  });
+
+  ok(firstField, 'No fields in query');
+  const fields: string[] = firstField.split('_');
+  ok(fields.length === 2, `expected 2 elements in ${fields}`);
+
+  const namespace = fields[0];
+  const name = singular(fields[1]);
```

Comment:
if we have `person` and `people`, what would happen?

Reply:
🤯

Reply:
It looks like they handle it actually - https://github.com/plurals/pluralize/blob/cba2fef5aacec60dc1ff4c780d01b16bcb8a03b3/test.js#L54